File: examples/pizzas_static_ruby/lib/pizzas_domain/pizza/events/created_pizza.rb
module PizzasDomain
  class Pizza
    module Events
      class CreatedPizza
        attr_reader :aggregate_id, :name, :description, :toppings, :occurred_at

        def initialize(aggregate_id: nil, name: nil, description: nil, toppings: nil)
          @aggregate_id = aggregate_id
          @name = name
          @description = description
          @toppings = toppings
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: pizzas_static_go/domain/created_pizza.go
package domain

import "time"

type CreatedPizza struct {
	AggregateID string `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	Name string `json:"name"`
	Description string `json:"description"`
	Toppings []Topping `json:"toppings"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (c CreatedPizza) EventName() string { return "CreatedPizza" }

func (c CreatedPizza) GetOccurredAt() time.Time { return e.OccurredAt }
